packages/aws-otel-collector/aws-otel-collector.spec

```diff
@@ -0,0 +1,77 @@
+%global goproject github.com/aws-observability
+%global gorepo aws-otel-collector
+%global goimport %{goproject}/%{gorepo}
+
+Name: %{_cross_os}aws-otel-collector
+Version: 0.40.0
+Release: 1%{?dist}
+Summary: AWS Distro for OpenTelemetry Collector
+License: Apache-2.0 AND BSD-2-Clause AND BSD-3-Clause AND MIT AND MPL-2.0
+URL: https://github.com/aws-observability/aws-otel-collector
+Source0: %{gorepo}-v%{version}.tar.gz
+Source1: aws-otel-collector.service
+Source2: aws-otel-collector-tmpfiles.conf
+Source3: aws-otel-collector.yaml
+
+# Change log file path from /opt to /var/log
+Patch0001: 0001-logger-change-collector-log-file-path.patch
+
+BuildRequires: %{_cross_os}glibc-devel
+Requires: %{name}(binary)
+
+%description
+%{summary}.
+
+%package bin
+Summary: Telemetry collector binary
+Provides: %{name}(binary)
+Requires: (%{_cross_os}image-feature(no-fips) and %{name})
+Conflicts: (%{_cross_os}image-feature(fips) or %{name}-fips-bin)
+
+%description bin
+%{summary}.
+
+%package fips-bin
+Summary: Telemetry collector binary, FIPS edition
+Provides: %{name}(binary)
+Requires: (%{_cross_os}image-feature(fips) and %{name})
+Conflicts: (%{_cross_os}image-feature(no-fips) or %{name}-bin)
+
+%description fips-bin
+%{summary}.
+
+%prep
+%autosetup -n %{gorepo}-%{version} -p1
+
+%build
+export GO_MAJOR="1.22"
+
+%set_cross_go_flags
+
+go build -ldflags "${GOLDFLAGS}" -o aws-otel-collector ./cmd/awscollector
+gofips build -ldflags "${GOLDFLAGS}" -o fips/aws-otel-collector ./cmd/awscollector
+
+%install
+install -d %{buildroot}%{_cross_tmpfilesdir}
+install -p -m 0644 %{S:2} %{buildroot}%{_cross_tmpfilesdir}
+
+install -D -p -m 0644 %{S:1} %{buildroot}%{_cross_unitdir}/aws-otel-collector.service
+
+install -d %{buildroot}{%{_cross_bindir},%{_cross_fips_bindir}}
+install -p -m 0755 aws-otel-collector %{buildroot}%{_cross_bindir}
+install -p -m 0755 fips/aws-otel-collector %{buildroot}%{_cross_fips_bindir}
+
+install -d %{buildroot}%{_cross_factorydir}%{_cross_sysconfdir}
+install -p -m 0644 %{S:3} %{buildroot}%{_cross_factorydir}%{_cross_sysconfdir}
+
+%files
+%{_cross_attribution_file}
+%{_cross_unitdir}/aws-otel-collector.service
+%{_cross_tmpfilesdir}/aws-otel-collector-tmpfiles.conf
+%{_cross_factorydir}%{_cross_sysconfdir}/aws-otel-collector.yaml
```

Comment:
FWIW, I've thought about two options for downstreams to override our default configurations files for those service that don't have an API like this or the SSM agent:

- Override `Exec` directive in systemd services to force the service to look at a different location where the actual configuration file provided by the downstream resides
- Use a `tmpfiles.d` configuration file that installs the downstream overrides at the location where the service will look at: if `tmpfiles.d` detects that multiple configuration files provide the same `target` file, if will user lexicographic order to determine which one to keep and ultimately create

I think we could suggest downstreams to follow the latter, unless others have a better proposal to allow dowstreams override the defaults we provide. (cc: @bcressey , what are your thoughts on this?)

Reply:
I am a fan of `tmpfiles.d` as the clean, simple way to override a default here. It's my uneducated opinion, based on working with `tmpfiles.d` on other packages for configuration things that live in `/etc/something.d`.

Reply:
> I think we could suggest downstreams to follow the latter, unless others have a better proposal to allow dowstreams override the defaults we provide.

@arnaldo2792 the tmpfiles behavior here is documented but surprising to me (emphasis added):
> If multiple files specify the same path, the entry in the file with the lexicographically **earliest** name will be applied

Elsewhere, for drop-ins and such, the latest file in sorted order takes effect. So this is likely something downstream builds will get wrong, or we'll add a config to a package here and break them if our tmpfiles config has an earlier filename.

You could make the config file path an environment variable, and then downstreams could override just that rather than the entire `Exec` directive using a drop-in.

It'd be annoying to implement, but we could prototype making Bottlerocket's `/etc` an overlayfs, consisting of a read-only set of files under the sys-root, and a tmpfs upper directory. Then packages could install files directly to `/etc`, and downstreams could clobber them with a tmpfiles directive:
```
r /etc/aws-otel-collector.yaml
C /etc/aws-otel-collector.yaml - - - - -
```

Reply:
I had an offline conversation with @bcressey , and we agreed that copying the default config provided by us to `/etc` is OK, and that downstreams can override the config if they provide their own API server services that override the configurations if they need to.